### ISSUE TITLE
Autocomplete not working?

### ISSUE THREAD

**Issue by felixfbecker** [felixfbecker/php-language-server#276]
_From @snizemic on February 4, 2017 7:50_

I don't know if I am doing something wrong or if it is even possible but I have framework classes in vendor folder so I would thought that autocomplete function hinting/completing these classes, eg. in use statement, function parameters etc. Every other function of your extension works well, great job! But autocomplete is not, even with your PDO example.

_Copied from original issue: felixfbecker/vscode-php-intellisense#67_

**Comment by felixfbecker** [comment]
Cannot reproduce. Any errors logged in Help > Toggle developer tools? Have you tried the autocomplete fixtures that are used in the tests?

**Comment by snizemic** [comment]
There is no error logged. But I was apparently wrong. If I open pure php file wihthout class definiton in it, the completion works well. But if I type anything in class definition nothing happens. I expect to show available attributes and methods (API) of current variable or statement. You know, as you can see in classic IDE. Maybe it is more code hinting than autocomplete. I hope you understand me :-)

**Comment by felixfbecker** [comment]
Can you please give me a snippet I can reproduce with

**Comment by snizemic** [comment]
Of course. Just try `composer create-project nette/web-project` and then go to `app/presenters/HomepagePresenter.php` and try anything there, eg. if you want to add `use Nette\Database\Context`, I thought that if I type `Nette\` and then press `Ctrl + Space`, the classes from namespace `Nette` (from vendor folder) will be shown (hinting).
But as I said maybe I am wrong and expect something what could not be done. I really appreciate your interest, thanks!

**Comment by rhyek** [comment]
I don't understand how autocomplete works with the vs code extension. I'm trying it out with a new Laravel project and nothing is suggested anywhere. Other functions work correctly.

**Comment by felixfbecker** [comment]
@snizemic @rhyek Some cases may not work. But there are tests for the general cases, you can see some scenarios in the fixtures folder and also in the demo gif.

**Comment by bkonetzny** [comment]
Same issue here, no completions at all (even the PDO from example is not working). No log entry about errors in console. Is there any way to help debug this? Running a Yii 1.x project not managed by composer, but php class files are indexed according to the log.

**Comment by Shoot-tigers-tonight** [comment]
PHP version 7.0.17

View the process and find this command

/private/var/folders/m2/w93r7ymj48gcw1qk7m6tjp9r0000gn/T/q6RhmQ /Users/tiger/.vscode/extensions/felixfbecker.php-intellisense-1.1.3/vendor/felixfbecker/language-server/bin/php-language-server.php

The error message

![image](https://cloud.githubusercontent.com/assets/6560524/24390158/9073f17c-13b9-11e7-965a-be85ac7ec474.png)

But not specified this config `/private/var/folders/m2/w93r7ymj48gcw1qk7m6tjp9r0000gn/T/q6RhmQ`

The use of the default, is not error.
Is this caused by this problem?

**Comment by weirdan** [comment]
I think I might be having the same problem, and it appears to be related to OS output buffering. Here's what I tried:

1. created new folder and opened it in vscode
1. created new file and typed:
    ```
    <?php
    
    $q = new PDO('z');
    $q->
    ```
1. observed responses from language server in developer console
1. there's a truncated response with `Content-Length: 4466` (the threshold is likely to be 4096 bytes, including headers), ending with `"sortText":null,"filterText":null,"insertText":null,"textEdit":null,"additiona` (note the truncated json)
1. waited for 10 minutes to see if this is just my computer being slow. Nope.
1. Hovered the `PDO` class name
1. Bamm! Both autocomplete window and hover hint would appear.
1. Developer console shows that the remaining part of that truncated response arrived, along with the hint response (which is much shorter):
    ```
    ... "textEdit":null,"additiona

    [Extension Host] lTextEdits":null,"command":null,"data":null},{"label":"__wakeup","kind":2,"detail":"mixed","documentation":null,"sortText":null,"filterText":null,"insertText":null,"textEdit":null,"additionalTextEdits":null,"command":null,"data":null},{"label":"__sleep","kind":2,"detail":"mixed","documentation":null,"sortText":null,"filterText":null,"insertText":null,"textEdit":null,"additionalTextEdits":null,"command":null,"data":null}]},"id":11,"jsonrpc":"2.0"}Content-Type: application/vscode-jsonrpc; charset=utf8
    Content-Length: 234

    {"result":{"contents":[{"language":"php","value":"<?php\nclass PDO  {"},"Represents a connection between PHP and a database server."],"range":{"start":{"line":2,"character":9},"end":{"line":2,"character":12}}},"id":12,"jsonrpc":"2.0"}
    ```

Versions:
* vscode = 1.14
* php intellisense extension = 1.4.3
* php language server = v4.5.3
* php = 7.1.6-2 (Debian)

**Comment by MadHed** [comment]
@weirdan this is probably the same issue: https://github.com/felixfbecker/php-language-server/issues/435
I couldn't reproduce this ony any other machine so far. If you could give any more clues, that would be great!

**Comment by weirdan** [comment]
@MadHed right, it certainly is.

**Comment by apt425410** [comment]
![image](https://user-images.githubusercontent.com/27874509/48908372-a546f580-ee9c-11e8-85f4-a786c5d6e176.png)

i don't know any clue
